File: code wars/7kyu/Lottery machine/main.py
import re

def lottery(s:str) -> str:
    match =  re.findall('\d', s)
    return ''.join(dict.fromkeys(match)) if match else 'One more run!'

# Without Regexp
def lottery(s):
    return "".join(dict.fromkeys(filter(str.isdigit, s))) or "One more run!"

File: code wars/7kyu/Lottery machine/main.js
// With regexp
function lottery(str){
    let arr = str.match(/[0-9]/g);
    return arr ? [...new Set(arr)].join('') : 'One more run!';
 }
// can be shorten with
const lot =  str => (arr = str.match(/[0-9]/g)) ? [...new Set(arr)].join('') : 'One more run!';
// Without regexp
const lottery = str => {
    return [...new Set([...str])].filter(e => e >= 0 && e <= 9).join('') || 'One more run!';
}